synapse/lib/base.py

```diff
@@ -45,25 +58,42 @@ class Base:
 
     Example:
 
-        class Foo(Anit):
+        class Foo(Base):
 
-            def __init__(self, x):
-                self.x = x
+            async def __anit__(self, x, y):
 
-            async def __anit__(self):
-                await stuff()
+                await Base.__anit__(self)
+
+                await stuff(x, y)
 
         foo = await Foo.anit(10)
+
+    Note:
+        One should not create instances directly via its initializer, i.e. Base().  One shall always use the class
+        method anit.
     '''
+    def __init__(self):
+        assert inspect.stack()[1].function == 'anit', 'Objects from Base must be constructed solely via "anit"'
```

Comment:
What happens here if this statement is optimized away? It feels like it should be a strong assertion; not a `assert` statement.